src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -579,7 +581,9 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		this.inputEditor.setPosition({ lineNumber: 1, column: value.length + 1 });
 
 		if (!transient) {
-			this.saveCurrentValue(this.getInputState());
+			const inputState = this.getInputState();
+			inputState.chatContextAttachments = inputState.chatContextAttachments?.filter(attachment => !attachment.isImage);
+			this.saveCurrentValue(inputState);
```

Comment:
Should the filtering happen inside saveCurrentValue instead of doing it in multiple places?

Reply:
I moved it out since it led to some type inference since the paramater of `saveCurrentValue` is `any`, but I think that works too